### ISSUE TITLE
[TF2] Random Crash Matqueue Segfault

### ISSUE THREAD

**Issue by ibrokemypie** [ValveSoftware/Source-1-Games#2310]
Getting completely random crashes in tf2 and the only message anywhere is in dmesg, stating ``MatQueue0[1029]: segfault at 0 ip 00000000f73eb2d9 sp 0000000008519a18 error 6 in libc-2.25.so[f72b8000+1b9000]
``

**Comment by kisak-valve** [comment]
Hello @ibrokemypie, please copy your system information from steam (```Steam``` -> ```Help``` -> ```System Information```) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. In particular, if you can find any CrashIDs or minidumps related to this issue, please include them.

**Comment by ibrokemypie** [comment]
Gist: https://gist.github.com/ibrokemypie/d2df59ac16f4c0b8b9cffd5a156e4831

No dumps that I can find

**Comment by ibrokemypie** [comment]
MatQueue0[8758]: segfault at 0 ip 00000000f742e2d9 sp 0000000006bc1a18 error 6 in libc-2.25.so[f72fb000+1b9000]

**Comment by mastercoms** [comment]
Try `r_queued_decals 0`.

**Comment by ibrokemypie** [comment]
in the game console?

**Comment by kisak-valve** [comment]
Crashing in i965_dri.so hints that this is a video driver issue, and should be brought to the attention of the driver vendor if it has not been already. Also, this could just be a variation of #2212.

**Comment by mastercoms** [comment]
What is the value of `mat_queue_mode`?

**Comment by ibrokemypie** [comment]
@mastercoms -1

**Comment by ibrokemypie** [comment]
```
[11727.500015] MatQueue0[2760]: segfault at 0 ip 00000000f4a9cbd6 sp 0000000008254180 error 6 in i965_dri.so[f469d000+7a7000]
[12060.654944] MatQueue0[12526]: segfault at 0 ip 00000000f73dc2d9 sp 000000000499c9b8 error 6 in libc-2.25.so[f72a9000+1b9000]

```

**Comment by ibrokemypie** [comment]
this is also now occuring with amdgpu on my new vega 56, linux 4.15

**Comment by nightsky30** [comment]
@ibrokemypie , Opened a ticket here:  https://bugs.freedesktop.org/show_bug.cgi?id=103810
Might be the same issue.  Do you know exactly when and what version of Mesa drivers might have started causing the error?  I saw in your gist you had v17.1.5.  Had you noticed these errors after a system update, or had you been running 17.1.5 for a while?

**Comment by ibrokemypie** [comment]
@nightsky30 I dont know what version and currently am unable to diagnose as getting other crashes now before this one (regarding out of memory)
If it helps, I am currently running 17.3.1

**Comment by nightsky30** [comment]
This appears to be fixed for me with the latest Mesa Git 18.0.1

[System Info](https://gist.github.com/nightsky30/a36a2a69e488c55894bd50142948adc0)

**Comment by kisak-valve** [comment]
Thanks for retesting, closing.
